src/commands/delete_command.hpp

```diff
@@ -0,0 +1,106 @@
+#ifndef DELETE_COMMAND_HPP
+#define DELETE_COMMAND_HPP
+
+#include "interactive_command_base.hpp"
+#include "command_manager.hpp"
+#include "../core/rq/RegionQueryService.hpp"
+#include "direct_command_base.hpp"
+
+class dicmdDeleteObj : public DirectCommandBase
+{
+public:
+	dicmdDeleteObj(IObjectPoolPtr ptr, QPoint pos)
+		: m_workingSet(std::dynamic_pointer_cast<WorkingSet>(ptr))
+	{
+		add_option("-point",new PointCommandOptionValue(pos));
+	}
+
+	dicmdDeleteObj(IObjectPoolPtr ptr)
+		: m_workingSet(std::dynamic_pointer_cast<WorkingSet>(ptr))
+	{
+		add_option("-point",new PointCommandOptionValue());
+	}
+
+	virtual std::string get_name() override
+	{
+		return "dicmdDeleteShape";
+	}
+
+	virtual void execute() override
+	{
+		// find object from RQ
+		//FIXME need only one init during shape creation
+		RegionQuery& rq = RegionQuery::getInstance();
```

Comment:
Will be done via merge from viewport_operations.
Adding from tree will be done from directive shape creation commands on successful commit.